### PR TITLE
chore: simplify Registry api

### PR DESCRIPTION
**Pull request by dgozman**

### CODE REVIEW

File: src/server/webkit/wkPage.ts [hunk 1]
```diff
@@ -768,14 +768,13 @@ export class WKPage implements PageDelegate {
 
   private _toolbarHeight(): number {
     if (this._page._browserContext._browser?.options.headful)
-      return hostPlatform.startsWith('10.15') ? 55 : 59;
+      return registry.hostPlatform() === 'mac10.15' ? 55 : 59;
```
Comment (yury-s) [inline src/server/webkit/wkPage.ts:771]: Move `hostPlatform` to from registry.ts to utils.ts

File: src/utils/browserFetcher.ts [hunk 2]
```diff
@@ -52,8 +50,8 @@ export async function downloadBrowserWithProgressBar(registry: Registry, browser
     progressBar.tick(delta);
   }
 
-  const url = registry.downloadURL(browserName);
-  const zipPath = path.join(os.tmpdir(), `playwright-download-${browserName}-${hostPlatform}-${registry.revision(browserName)}.zip`);
+  const url = downloadURL;
+  const zipPath = path.join(os.tmpdir(), `${downloadFileName}.zip`);
```
Comment (yury-s) [inline src/utils/browserFetcher.ts:54]: Is there a reason downloadFileName doesn't include `.zip` extension?

File: src/utils/registry.ts [hunk 3]
```diff
@@ -248,38 +245,46 @@ export function isBrowserDirectory(browserDirectory: string): boolean {
...
-        installByDefault: !!obj.installByDefault,
-        // Method `isBrowserDirectory` determines directory to be browser iff
-        // it starts with some browser name followed by '-'. Some browser names
-        // are prefixes of others, e.g. 'webkit' is a prefix of `webkit-technology-preview`.
-        // To avoid older registries erroneously removing 'webkit-technology-preview', we have to
-        // ensure that browser folders to never include dashes inside.
-        browserDirectory: browserDirectoryPrefix.replace(/-/g, '_') + '-' + revision,
-      };
-    });
+  hostPlatform(): BrowserPlatform {
```
Comment (yury-s) [inline src/utils/registry.ts:286]: drop this from registry

**Review by yury-s — APPROVED** [review]

File: src/utils/browserFetcher.ts (hunk 2 above)
Comment (dgozman) [reply src/utils/browserFetcher.ts:54]: Not really!

**Pull request merged**